mqtt/mqtt-broker/src/configuration/mod.rs

```diff
@@ -0,0 +1,190 @@
+mod size;
+
+pub use size::HumanSize;
+
+use std::{
+    convert::From,
+    num::NonZeroUsize,
+    path::{Path, PathBuf},
+    time::Duration,
+};
+
+use config::{Config, ConfigError, File, FileFormat};
+use serde::Deserialize;
+
+pub const DEFAULTS: &str = include_str!("../../config/default.json");
+
+#[derive(Debug, Clone, Deserialize)]
+#[serde(rename_all = "snake_case")]
+pub enum Transport {
+    Tcp {
+        address: String,
+    },
+    Tls {
+        address: String,
+        certificate: PathBuf,
+    },
+}
+
+#[derive(Debug, Deserialize, Clone, Copy, PartialEq)]
+#[serde(rename_all = "snake_case")]
+pub enum QueueFullAction {
+    DropNew,
+    DropOld,
+}
+
+#[derive(Debug, Deserialize, Clone)]
+pub struct RetainedMessages {
+    max_count: u32,
+    #[serde(with = "humantime_serde")]
+    expiration: Duration,
+}
+
+#[derive(Debug, Deserialize, Clone)]
+pub struct SessionPersistence {
+    file_path: String,
+    #[serde(with = "humantime_serde")]
+    time_interval: Duration,
+    unsaved_message_count: u32,
+}
+
+#[derive(Debug, Deserialize, Clone, PartialEq)]
+pub struct SessionConfig {
+    #[serde(with = "humantime_serde")]
+    expiration: Duration,
+    max_message_size: Option<HumanSize>,
+    max_inflight_messages: usize,
+    max_queued_messages: usize,
```

Comment:
What if this is not specified in the config? It will default to 0 (= no limit)?

Reply:
we have `"max_queued_messages": 1000` and `"max_queued_size": null` (=no limits) by default